File: ProjecTree-was/src/main/java/com/ssafy/projectree/domain/tech/model/repository/NodeTechStackRepository.java
package com.ssafy.projectree.domain.tech.model.repository;

import com.ssafy.projectree.domain.node.model.entity.Node;
import com.ssafy.projectree.domain.tech.model.entity.NodeTechStack;
import io.lettuce.core.dynamic.annotation.Param;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface NodeTechStackRepository extends JpaRepository<NodeTechStack, Long> {
    List<NodeTechStack> findAllByNode(Node node);

    @Modifying(clearAutomatically = true)
    @Query("""
        update TechStackInfo tsi
        set tsi.isSelected = false
        where tsi.id in (
            select nts.techStackInfo.id
            from NodeTechStack nts
            where nts.node.id = :nodeId
        )
    """)
    void unselectAllByNodeId(@Param("nodeId") Long nodeId);
}

File: INFERENCE/app/db/repository/node_tech_stack_repository.py
from app.db.repository.base_repository import BaseRepository
from app.db.models import NodeTechStack
from app.db.schemas.tech import NodeTechStackCreate, NodeTechStackUpdate


class NodeTechStackRepository(
    BaseRepository[NodeTechStack, NodeTechStackCreate, NodeTechStackUpdate]
):
    pass


node_tech_stack_repository = NodeTechStackRepository(NodeTechStack)
